modules/packages/ProtobufProtocolSupport.chpl

```diff
@@ -0,0 +1,881 @@
+/*
+ * Copyright 2020 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/* Documentation for ProtobufProtocolSupport */
```

Comment:
@lydia-duncan should we document the parent module and the submodules separately? No function in these modules is user exposed.

Reply:
Do you mean "should we add comments?" or "should we add `pragma "no doc"` to these symbols?"  Or both?  It also doesn't look like the parent module provides anything beyond its three submodules - if the intention is for the submodules to be merely organizational while most places that need it would just import or use the parent module, then I think it's fine for the summary documentation to be on the parent rather than the particular submodules (though I do think there should be some information on why the submodules were divided in the way they were, to help developers determine where to place future functions).

Reply:
> I do think there should be some information on why the submodules were divided in the way they were, to help developers determine where to place future functions).

Adding comments for that would be sufficient?

Reply:
> Do you mean "should we add comments?" or "should we add `pragma "no doc"` to these symbols?" Or both?

I don't think comment for each function will be required, they are self-explanatory by name as long as we have explanatory documentation for the modules, but yeah should we add pragma "no doc"?

Reply:
> I think you could get away with adding it to the module symbols and having it propagate that way.

By this did you mean something like this-

```
pragma "no doc"
module RepeatedFields {
```

Reply:
And also we should have detailed documentation for the parent module, explaining all the submodules right?

Reply:
Sorry, meant to come back to this before my vacation.  Yeah, I think the documentation for the parent module should explain the division into the submodules if the submodules themselves do not get documented